src/loop.c

```diff
@@ -91,3 +92,31 @@ int main() {
     loop(epoll_c, events);
   }
 }
+
+void read_message(int file_descriptor) {
+  int message_len = full_message_availiable(file_descriptor);
+
+  if (message_len) {
+    uint8_t message[PIECE_SIZE_BYTES +
+                    8]; // The largest valid message size available
```

Comment:
it's probably clearer to reuse the same constant everywhere to avoid confusion 
```suggestion
    uint8_t message[MAX_SIZE_MESSAGE_INT*4];
```